src/runtime/relax_vm/lm_support.cc

```diff
@@ -510,6 +511,72 @@ int SampleTopPFromProb(NDArray prob, double top_p, double uniform_sample) {
 
 TVM_REGISTER_GLOBAL("vm.builtin.sample_top_p_from_prob").set_body_typed(SampleTopPFromProb);
 
+NDArray LogProbsFromLogits(NDArray logits) {
```

Comment:
Just call it logsoftmax

Reply:
Consider pass in the output NDArray, so they can be pre-allocated

Reply:
Do you mean I need to pre-allocate output NDArray and pass it as function arg?